plugins/pixart-tp/fu-pxi-tp-firmware.c

```diff
@@ -0,0 +1,953 @@
+/*
+ * Copyright 2025 Harris Tai <harris_tai@pixart.com>
+ * Copyright 2025 Micky Hsieh <micky_hsieh@pixart.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later
+ */
+
+// =============================================================
+// file: fu-pxi-tp-firmware.c (pure C, offset-based parsing; no packed structs)
+// =============================================================
```

Comment:
@PixartHarris no // comments please -- and this kind of comment isn't really useful either